lighthouse-core/gather/driver.js

```diff
@@ -143,17 +143,33 @@ class Driver {
         (_ => reject(new Error('The asynchronous expression exceeded the allotted time of 60s'))),
         60000
       );
+
       this.sendCommand('Runtime.evaluate', {
-        expression: asyncExpression,
+        expression: `(function wrapInNativePromise() {
+          const __nativePromise = window.__nativePromise || Promise;
+          return new __nativePromise(function(resolve) {
+            const wrapError = ${wrapRuntimeEvalErrorInBrowser.toString()};
+            try {
+              __nativePromise.resolve(${asyncExpression}).then(resolve, wrapError);
+            } catch (e) {
+              wrapError(e);
+            }
+          });
+        }())`,
         includeCommandLineAPI: true,
         awaitPromise: true,
         returnByValue: true
       }).then(result => {
         clearTimeout(asyncTimeout);
+        const value = result.result.value;
+
         if (result.exceptionDetails) {
-          reject(result.exceptionDetails.exception.value);
+          // An error occurred before we could even enter our try block, should be *very* rare
+          reject(new Error('an unknown driver error occurred'));
+        } if (value && value.__failedInBrowser) {
+          reject(Object.assign(new Error(), value));
         } else {
-          resolve(result.result.value);
+          resolve(value);
```

Comment:
yeah, I think @paulirish is addressing this in his latest comments in #941. We basically need a way to say "this is an error I expected (fetch rejected on offline request or whatever)" vs "whoooops". For this, the caller of `driver.evaluateAsync` will have to tell the difference for now and we can revisit as we figure out error paths